Fix failing pre-commits after making 3.7 default

potiuk opened the pull request:
In order to keep consistency, both mypy and flake were always
using Python 3.6 - simply to make sure that the same errors
are produced for everyone. When default python version was
changed in #19317 the regular PR static checks started to fail
because Python 3.6 image is no longer built for "regular" PRs
(it's only built when "full-tests-needed" is set or when
the build runs in "main" as result of push or schedule).

Unfortunately those "defaults" could not be read from
"default python version" easily, because they are also
executed in pre-commits, which do not have the same
initialization as `breeze` or other CI scripts so there
are literally 3 more places where Python 3.6 should be
changed to 3.7.

This PR fixes it.

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

potiuk committed:
Fix failing pre-commits after making 3.7 default

In order to keep consistency, both mypy and flake were always
using Python 3.6 - simply to make sure that the same errors
are produced for everyone. When default python version was
changed in #19317 the regular PR static checks started to fail
because Python 3.6 image is no longer built for "regular" PRs
(it's only built when "full-tests-needed" is set or when
the build runs in "main" as result of push or schedule).

Unfortunately those "defaults" could not be read from
"default python version" easily, because they are also
executed in pre-commits, which do not have the same
initialization as `breeze` or other CI scripts so there
are literally 3 more places where Python 3.6 should be
changed to 3.7.

This PR fixes it.

uranusjr commented on airflow/api_connexion/endpoints/dag_endpoint.py:
```diff
@@ -111,15 +110,15 @@ def patch_dag(session, dag_id, update_mask=None):
 @security.requires_access([(permissions.ACTION_CAN_DELETE, permissions.RESOURCE_DAG)])
 @provide_session
-def delete_dag(dag_id: str, session: Session):
+def delete_dag(dag_id: str, session):
```
I don’t think this is supposed to fail?

uranusjr commented on airflow/cli/simple_table.py:
```diff
@@ -70,13 +70,13 @@ def print_as_plain_table(self, data: List[Dict]):
             return
         rows = [d.values() for d in data]
-        output = tabulate(rows, tablefmt="plain", headers=data[0].keys())
+        output = tabulate(rows, tablefmt="plain", headers=list(data[0].keys()))
```
```suggestion
        output = tabulate(rows, tablefmt="plain", headers=list(data[0]))
```

uranusjr commented on airflow/config_templates/default_webserver_config.py:
```diff
@@ -18,12 +18,12 @@
 
-# from airflow.www.fab_security.manager import AUTH_LDAP
-# from airflow.www.fab_security.manager import AUTH_OAUTH
-# from airflow.www.fab_security.manager import AUTH_OID
-# from airflow.www.fab_security.manager import AUTH_REMOTE_USER
+# from airflow.www.fab_security.manager import AUTH_LDAP  # type: ignore
+# from airflow.www.fab_security.manager import AUTH_OAUTH  # type: ignore
+# from airflow.www.fab_security.manager import AUTH_OID  # type: ignore
+# from airflow.www.fab_security.manager import AUTH_REMOTE_USER  # type: ignore
```
Why?

uranusjr commented on airflow/kubernetes/kubernetes_helper_functions.py:
```diff
@@ -84,4 +84,8 @@ def annotations_to_key(annotations: Dict[str, str]) -> Optional[TaskInstanceKey]
             .scalar()
         )
 
+    if not run_id:
+        # TODO: Is this the right way of handling it?
+        raise ValueError("Run Id has to be defined either by `run_id` annotation "
+                         "or derived from `execution_Date")
```
We can also just do

```python
(run_id,) = (
    session.query(TaskInstance.run_id)
    ...
    .one()
)
```

and let SQLAlchemy raise when there’s not a matching task instance found (it shouldn’t happen).

uranusjr reviewed (commented):
Didn’t read everything but there are some things we can do a bit better

potiuk committed:
Re-enables MyPy in non-failure mode

This PR enables mypy back as pre-commit for local changes after
the #19317 switched to Python 3.7 but also it separates out
mypy to a separate non-failing step in CI.

In the CI we will be able to see remaining mypy errors.

This will allow us to gradually fix all the mypy errors and enable
mypy back when we got all the problems fixed.
